### PR TITLE
KAFKA-9748: Extend Streams integration tests for EOS beta

### PR DESCRIPTION
**Pull request by mjsax**
The original PR was only considering `EosIntegrationTest`. However, there are actually a few more integration test with eos-alpha enabled that we missed to update.

Call for review @abbccdda @guozhangwang

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/integration/GlobalKTableEOSIntegrationTest.java [hunk 1]
```diff
@@ -319,15 +334,9 @@ private void produceAbortedMessages() throws Exception {
     }
 
     private void produceInitialGlobalTableValues() throws Exception {
-        produceInitialGlobalTableValues(true);
-    }
-
-    private void produceInitialGlobalTableValues(final boolean enableTransactions) throws Exception {
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/GlobalKTableEOSIntegrationTest.java:325]: We always set it to `true` -- removing unnecessary code.

File: streams/src/test/java/org/apache/kafka/streams/state/internals/StreamThreadStateStoreProviderTest.java [hunk 2]
```diff
@@ -361,7 +360,7 @@ private StreamTask createStreamsTask(final StreamsConfig streamsConfig,
         final ProcessorStateManager stateManager = new ProcessorStateManager(
             taskId,
             Task.TaskType.ACTIVE,
-            EXACTLY_ONCE.equals(streamsConfig.getString(StreamsConfig.PROCESSING_GUARANTEE_CONFIG)),
+            StreamThread.eosEnabled(streamsConfig),
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/StreamThreadStateStoreProviderTest.java:363]: Side cleanup

File: streams/src/test/java/org/apache/kafka/streams/integration/RocksDBMetricsIntegrationTest.java [hunk 3]
```diff
@@ -101,8 +101,9 @@
     @Parameters(name = "{0}")
     public static Collection<Object[]> data() {
         return Arrays.asList(new Object[][] {
+            {StreamsConfig.AT_LEAST_ONCE},
             {StreamsConfig.EXACTLY_ONCE},
-            {StreamsConfig.AT_LEAST_ONCE}
+            {StreamsConfig.EXACTLY_ONCE_BETA}
         });
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/RocksDBMetricsIntegrationTest.java:107]: General question: we record the same metric for all three processing modes. Why do you need to parametrize this test? \cc @cadonna

File: streams/src/test/java/org/apache/kafka/streams/integration/KTableSourceTopicRestartIntegrationTest.java [hunk 4]
```diff
@@ -130,7 +130,7 @@ public void shouldRestoreAndProgressWhenTopicWrittenToDuringRestorationWithEosDi
     }
 
     @Test
-    public void shouldRestoreAndProgressWhenTopicWrittenToDuringRestorationWithEosEnabled() throws Exception {
+    public void shouldRestoreAndProgressWhenTopicWrittenToDuringRestorationWithEosAlphaEnabled() throws Exception {
```
Comment (abbccdda) [inline streams/src/test/java/org/apache/kafka/streams/integration/KTableSourceTopicRestartIntegrationTest.java:133]: Could we extract the test portion as a common function?

**Review by abbccdda — COMMENTED** [review]
LGTM, with a comment

**Comment by mjsax** [comment]
Both Jenkins runs passed. Magic!

**Review by abbccdda — APPROVED** [review]
LGTM! Merge after jenkins passed

**Comment by mjsax** [comment]
Build failed with
```
23:01:46 [ant:checkstyle] [ERROR] /home/jenkins/jenkins-slave/workspace/kafka-pr-jdk11-scala2.13@2/clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java:332:5: NPath Complexity is 588 (max allowed is 500). [NPathComplexity]
```

Seems to be fixed via #8447

Can retest after more reviews.

**Review by guozhangwang — APPROVED** [review]
LGTM. Please feel free after green builds (maybe also run locally multiple times to capture any flakiness regression earlier) and comments addressed.

**Comment by mjsax** [comment]
Retest this please.

**Comment by mjsax** [comment]
Java 8 passed.
Java 11: `kafka.api.ConsumerBounceTest.testRollingBrokerRestartsWithSmallerMaxGroupSizeConfigDisruptsBigGroup`

Retest this please.

**Comment by mjsax** [comment]
Java 8 passed.
Java 11: `kafka.api.SaslGssapiSslEndToEndAuthorizationTest.testNoDescribeProduceOrConsumeWithoutTopicDescribeAcl`

Different test as before. Also run the tests locally multiple times without issues. Merging this.

**Pull request merged**